torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py

```diff
@@ -28,6 +35,22 @@
     "get_symmetric_quantization_config",
 ]
 
+_QUANT_CONFIG_TO_ANNOTATOR = {}
+
+
+def register_annotator(quantization_configs: List[QuantizationConfig]):
```

Comment:
why is this indexed by quantization_config? are we still planning to support module fqn/type based config or other existing features in QConfigMapping or do we just want to support global qconfig?

Reply:
So the way I imagined this to be used is like this:
qnnpack, i will rename this to xnnpack, quantizer has two configs:
symmetric_static_config
symmetric_dynamic_config.

THis decorator registers annotation function for each.

Now if I want to quantize a model with dynamic q linear for linear and static quant for the rest, I would do it like this with composable quantizer.
```
dq_linear_config = QNNPACKQuantizer.get_dynamic_linear_config()
dq_linear_quantizer = QNNPACKQuantizer().set_global(dq_linear_config)

static_qconfig = QNNPACKQuantizer.get_symmetric_config()
static_quantizer = QNNPACKQuantizer().set_global(static_config)

composed_quantizer = ComposableQuantizer([dq_linear_quantizer, static_quantizer])
```

We can alternatively do this via:
```
qnnpack_quantizer = QNNPACKQuantizer().set_global(static_config)
# following api doesnt yet exist
qnnpack_quantizer.set_module_type_config(torch.nn.Linear, QNNPACKQuantizer.get_dynamic_linear_config()
```

I think both are fine. RIght now I am doing this mainly because it felt somewhat easier to use, but I am open to just doing via `set_module_config` or similar API.

Reply:
Let me think a bit more. If it makes sense to just set_module_type_config api, then I will remove the config-to-annotator mapping part from the PR and follow up separately.

Reply:
Ok I thought about this and I dont think we should do

```
dq_linear_config = QNNPACKQuantizer.get_dynamic_linear_config()
dq_linear_quantizer = QNNPACKQuantizer().set_global(dq_linear_config)

static_qconfig = QNNPACKQuantizer.get_symmetric_config()
static_quantizer = QNNPACKQuantizer().set_global(static_config)

composed_quantizer = ComposableQuantizer([dq_linear_quantizer, static_quantizer])

```

Reply:
But for now I will use annotator api until we do this better